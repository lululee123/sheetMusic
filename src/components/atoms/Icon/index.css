.icon {
	display: inline-flex;
	align-items: center;
	justify-content: center;
	flex-shrink: 0;

	&.normal {
		width: 24px;
	}

	&.small {
		width: 16px;
	}

	&.clickable {
		cursor: pointer;
	}

	& > svg {
		width: 100%;
		height: auto;
	}
}
