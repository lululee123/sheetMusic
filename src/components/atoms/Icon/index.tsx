import React from 'react';
import classnames from 'classnames';

import { isExist } from 'util/helper';

import styles from './index.css';

interface IconProperty {
	className?: string;
	src: string;
	size?: 'normal' | 'small';
	onClick?: () => void;
}

const Icon: React.FC<IconProperty> = ({ className, src: Svg, size = 'normal', onClick }) => (
	<div
		className={classnames(styles.icon, className, isExist(size) && styles[size], {
			[styles.clickable]: isExist(onClick),
		})}
		role="button"
		tabIndex={0}
		onClick={onClick}
		onKeyPress={() => {}}
	>
		<Svg />
	</div>
);

export default Icon;
