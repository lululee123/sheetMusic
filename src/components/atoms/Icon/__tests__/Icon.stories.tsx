import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { select } from '@storybook/addon-knobs';

import Icon from 'components/atoms/Icon';

import QuestionIcon from 'images/icon/circle-question.inline.svg';

const stories = storiesOf('atoms/Icon', module);

stories.add('__interactive', () => (
	<Icon
		src={QuestionIcon}
		size={select('size', ['normal', 'small'], 'normal')}
		onClick={action('onClick')}
	/>
));
