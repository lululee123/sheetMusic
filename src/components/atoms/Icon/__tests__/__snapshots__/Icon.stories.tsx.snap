// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="icon normal clickable"
    role="button"
    tabindex="0"
  >
    <test-file-stub />
  </div>
</DocumentFragment>
`;
